arm9/src/unk_0200BB14.c

```diff
@@ -0,0 +1,1280 @@
+}
+
+THUMB_FUNC void FUN_0200BCB0(u32 *param0)
+{
+    FUN_0201C350(param0[3]);
+    FUN_0201D12C();
+    FUN_0201E08C();
+
+    if (param0[2] == 1)
+    {
+        FUN_02009FA0();
+    }
+}
+
+THUMB_FUNC void FUN_0200BCD0(struct UnkStruct_0200BB14_1 *param0, struct UnkStruct_0200BB14_2 *param1)
+{
+    param0->unk004--;
+    FreeToHeap(param1);
+}
+
+THUMB_FUNC void FUN_0200BCE0(struct UnkStruct_0200BB14_1 *param0, struct UnkStruct_0200BB14_2 *param1)
+{
+    FUN_0200BC40(param1);
+    FUN_0200BC4C(param1);
+    FUN_0200BC5C(param1);
+    FUN_0200BCD0(param0, param1);
+}
+
+THUMB_FUNC void FUN_0200BD04(u32 *param0)
+{
+    GF_ASSERT(param0[1] == 0);
+    FUN_0200BCB0(param0);
+    FreeToHeap(param0);
+}
+
+THUMB_FUNC u32 FUN_0200BD20(
+    struct UnkStruct_0200BB14_1 *param0, struct UnkStruct_0200BB14_2 *param1, const char **param2)
+{
+    int st14 = 6;
+    const char **st10 = param2;
+
+    if (param0 == NULL || param1 == NULL)
+    {
+        return 0;
+    }
+
+    if (param2[4] == 0)
+    {
+        st14 = 4;
+    }
+
+    param1->unk54 = st14;
+    u32 r2 = FUN_0200965C(st14);
+    param1->unk08 = AllocFromHeap(param0->unk000, r2 * st14);
+
+    for (int i = 0; i < st14; i++)
+    {
+        u32 st18 = FUN_02009660(param1->unk08, i);
+        void *st1c = FUN_020161A4(param0->unk000, st10[i]);
+
+        FUN_02009668(st1c, st18, param0->unk000);
+        FreeToHeap(st1c);
+    }
+
+    for (int i = 0; i < st14; i++)
+    {
+        param1->unk0c[i] =
+            FUN_02008DEC(FUN_020096CC(FUN_02009660(param1->unk08, i)), i, param0->unk000);
+    }
+
+    for (int i = 0; i < st14; i++)
+    {
+        u32 st20 = FUN_02009660(param1->unk08, i);
+        param1->unk24[i] = FUN_02009424(FUN_020096CC(st20), param0->unk000);
+        param1->unk3c[i] = FUN_020093A8(param1->unk0c[i], st20, param1->unk24[i], param0->unk000);
+    }
+
+    FUN_02009A90(param1->unk24[0]);
+    FUN_02009D34(param1->unk24[1]);
+    void *r6 = FUN_020161A4(param0->unk000, st10[6]);
+    param1->unk04 = FUN_02008BE0(r6,
+        param0->unk000,
+        param1->unk0c[0],
+        param1->unk0c[1],
+        param1->unk0c[2],
+        param1->unk0c[3],
+        param1->unk0c[4],
+        param1->unk0c[5]);
+    FreeToHeap(r6);
+
+    return 1;
+}
+
+THUMB_FUNC u32 FUN_0200BE38(u32 *param0, u32 *param1, struct UnkStruct_0200BB14_3 *param2)
+{
+    return FUN_0200BE74(param0,
+        param1,
+        param2->unk00,
+        param2->unk04,
+        param2->unk06,
+        param2->unk04,
+        param2->unk0a,
+        param2->unk0c,
+        param2->unk10,
+        param2->unk14,
+        param2->unk18,
+        param2->unk1c,
+        param2->unk20,
+        param2->unk24);
+}
+
+THUMB_FUNC u32 FUN_0200BE74(u32 *param0,
+    u32 *param1,
+    u32 param2,
+    s16 param3,
+    s16 param4,
+    s16 param5,
+    u16 param6,
+    u32 param7,
+    u32 param8,
+    u32 param9,
+    u32 param10,
+    u32 param11,
+    u32 param12,
+    u32 param13)
+{
+#pragma unused(param10)
```

Comment:
honestly I find this somewhat unlikely tbh, have you verified this by looking at what calls this?

Reply:
So this function is only used in the one just above which clearly put things on the stack. If they are not function arguments, I don't what they are

Reply:
local variables? they are also allocated to the stack